Modules/Misc/GameMenuButton.lua

```diff
@@ -179,22 +197,50 @@ function M:GameMenuButton()
         self.backgroundFade.specIcon:SetFont(iconsFont, F.FontSizeScaled(E.db.TXUI.addons.gameMenuSkin.specIconSize), "")
         self.backgroundFade.specIcon:SetTextColor(1, 1, 1, 1)
 
-        self.backgroundFade.guildText:SetText(guildName and F.String.FastGradientHex("<" .. guildName .. ">", "06c910", "33ff3d") or "")
+        self.backgroundFade.guildText:SetText(guildName and
+          F.String.FastGradientHex("<" .. guildName .. ">", "06c910", "33ff3d") or "")
         self.backgroundFade.specIcon:SetText(specIcon)
         self.backgroundFade.levelText:SetText("Lv " .. E.mylevel)
         self.backgroundFade.classText:SetText(F.String.GradientClass(E.myLocalizedClass, nil, true))
       end
 
-      if self.collections then self.collections.achievs:SetText("Achievement Points: " .. F.String.ToxiUI(E:FormatLargeNumber(GetTotalAchievementPoints(), ","))) end
+      if self.collections then
+        self.collections.achievs:SetText("Achievement Points: " ..
+          F.String.ToxiUI(E:FormatLargeNumber(GetTotalAchievementPoints(), ",")))
+      end
 
       if self.mythic then
+        -- Update M+ history
+        local history = C_MythicPlus.GetRunHistory(false, true)
+        local historyLimit = E.db.TXUI.addons.gameMenuSkin.historyLimit
+
+        for i = 1, historyLimit do
+          local historyFrame = self.mythic["history" .. i]
+          if historyFrame then
+            local historyRun = history[#history - i + 1]
+            local historyText
+            local historyTextPrefix = i .. ": "
+            if historyRun then
+              local historyDungeonName = C_ChallengeMode.GetMapUIInfo(historyRun.mapChallengeModeID)
+              historyText = historyRun.completed and
+                  F.String.Good(("%s (+%d)"):format(historyDungeonName, historyRun.level)) or
```

Comment:
Since `("%s (+%d)"):format(historyDungeonName, historyRun.level)` is identical for both cases, can we extract it?